sdks/typescript/client-typescript-eventsource/README.md

```diff
@@ -338,14 +346,34 @@ it is a header, the browser needs a parameter as the SSE spec doesn't allow for
 will automatically retrigger a refresh of your events if you have already connected (unless you are using polling
 and your polling interval is set to 0).
 
-To add a generic key/value pair, use `attribute_value(key, value)`, to use an array of values there is 
-`attribute_values(key, Array<value>)`. You can also `clear()`.
+**Sending custom attributes:**
+
+To add a custom key/value pair, use `attribute_value(key, value)`
+
+```typescript
+    featureHubRepository.clientContext.attribute_value('first-language', 'russian').build();
+```
+
+Or with array of values (only applicable to custom strategies):
+
+```typescript
+   featureHubRepository.clientContext.attribute_value('languages', ['russian', 'english', 'german']).build();
+```
+
+You can also `clear()`.
 
 In all cases, you need to call `build()` to re-trigger passing of the new attributes to the server for recalculation.
 
-By default, the _user key_ is used for percentage based calculations, and without it, you cannot participate in
-percentage based Rollout Strategies ("experiments"). However, a more advanced feature does let you specify other
-attributes (e.g. _company_, or _store_) that would allow you to specify your experiment on..
+
+**Coding for percentage splits:**
+For percentage rollout you are only required to provide the `userKey`. 
+
+```typescript
+    featureHubRepository.clientContext.userKey('ideally-unique-id').build(); 
+```
+
+For more details on percentage splits and feature experiments see [Percentage Split Rule](https://docs.featurehub.io/#_percentage_split_rule).
```

Comment:
session id should also be mentioned here if you are saying "required", because it isn't required. One of userkey or session is.